src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+import React, { Component } from 'react';
+
+import { TelemetryService } from 'services';
+import { DeviceIcon } from './deviceIcon';
+import { RulesGrid, rulesColumnDefs } from 'components/pages/rules/rulesGrid';
+import { translateColumnDefs } from 'utilities';
+import {
+  Flyout,
+  FlyoutHeader,
+  FlyoutTitle,
+  FlyoutCloseBtn,
+  FlyoutContent,
+  FlyoutSection
+} from 'components/shared';
+import {
+  PropertyGrid as Grid,
+  PropertyRow as Row,
+  PropertyHeaderCell as HeaderCell,
+  PropertyCell as Cell
+} from './propertyGrid';
+
+import './deviceDetails.css';
+
+export class DeviceDetails extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      alarms: undefined,
+      isAlarmsPending: false,
+      alarmsError: undefined
+    };
+    this.columnDefs = [
+      rulesColumnDefs.ruleName,
+      rulesColumnDefs.severity,
+      rulesColumnDefs.explore
+    ];
+  }
+
+  componentDidMount() {
+    if (!this.props.rulesLastUpdated) this.props.fetchRules();
+    this.fetchAlarms((this.props.device || {}).id);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if ((this.props.device || {}).id === nextProps.device.id) {
```

Comment:
This doesn't look right. It seems to be saying to fetch the alarms when the old and new device ids are the same. It should update when the ids are different. Right?